src/TLuaInterpreter.cpp

```diff
@@ -6262,15 +6262,13 @@ int TLuaInterpreter::setPopup(lua_State* L)
     int n = lua_gettop(L);
 
     // console name is an optional first argument
-    if (n > 4) {
+    if (n > 3) {
         windowName = WINDOW_NAME(L, s);
         s++;
     }
-    if (!lua_isstring(L, s)) {
-        lua_pushstring(L, "setPopup: wrong argument type");
-        return lua_error(L);
-    }
-    QString txt = lua_tostring(L, s);
+
+    // The next argument was a string, but it no longer does anything.
+    // Thus we simply skip it.
     s++;
```

Comment:
This could be simply rolled up into the next usage (in the following `if`) as `++s` - but for clarity I guess you can leave it here. However one should never use an isolated post-increment as that is less efficient that the pre-increment (as an extra memory location is needed to hold the old value for it to be used after the increment has been done) and unless one needs to use the post-increment value for something else at the same time one should prefer the pre-increment one.

tl;dr;
```suggestion
    ++s;
```

Reply:
If you think that any modern compiler creates different code for (stand-alone) pre- vs. post-increment, please think again; GCC generates the same code even under -O0, i.e. without optimization. Yes, I checked.